core/src/main/java/org/fao/geonet/kernel/security/listener/HttpSessionEventPublisher.java

```diff
@@ -0,0 +1,87 @@
+package org.fao.geonet.kernel.security.listener;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.fao.geonet.ApplicationContextHolder;
+import org.springframework.context.ApplicationContext;
+import org.springframework.security.web.session.HttpSessionCreatedEvent;
+import org.springframework.security.web.session.HttpSessionDestroyedEvent;
+import org.springframework.web.context.support.WebApplicationContextUtils;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpSessionEvent;
+import javax.servlet.http.HttpSessionListener;
+
+/* This is just a copy of the org.springframework.security.web.session.HttpSessionEventPublisher
```

Comment:
I updated the documentation to provide more details as to why this was required. It also references the bug that describes the issue.

This is required because our code was based on 4.8.3 which is the version that RedHat SSO 7.3 uses. This is our target software at the moment.  It was noticed that RedHat SSO has released a newer version that now uses the newer drivers. 
Our intention is to upgrade to the newer version of the software in the new year.  At that time I will create a PR to upgrade to newer version of the keycloak adapter and I can will be able to remove this file.